lib/datasource.js

```diff
@@ -1666,11 +1666,12 @@ DataSource.prototype.discoverSchemas = function(tableName, options, cb) {
       const propName = nameMapper('column', item.columnName);
       schema.properties[propName] = {
         type: item.type,
-        required: (item.nullable === 'N' || item.nullable === 'NO' ||
+        required: item.generated ? false : (item.nullable === 'N' || item.nullable === 'NO' ||
```

Comment:
Just to avoid ternary operator which causes cognitive complexity and difficult to understand.

```suggestion
        required: !item.generated && (item.nullable === 'N' || item.nullable === 'NO' ||
```